src/processes/follow-topic-process.js

```diff
@@ -31,26 +119,33 @@ const followTopicProcess = async (job, done) => {
         done(null, 'community message format is inactive or not found');
         return;
       }
-      // check user already received the message from admin
-      const chat = serverClient.channel('messaging', {
-        type_channel: 0,
-        members: [communityMessageFormat.user_id, data.user_id],
-        created_by_id: communityMessageFormat.user_id
-      });
-      await chat.create();
-
-      const channelState = await chat.watch();
 
-      if (channelState.messages.length === 0) {
-        await chat.sendMessage({
-          text: communityMessageFormat.message,
-          user_id: data.user_id
+      // check if user is anonymous
+      const senderUser = await User.findByPk(communityMessageFormat.user_id);
+      if (senderUser.is_anonymous) {
+        await sendMessageAsAnonymous(serverClient, communityMessageFormat, data);
+      } else {
+        // check user already received the message from admin
+        const chat = serverClient.channel('messaging', {
+          type_channel: 0,
+          members: [communityMessageFormat.user_id, data.user_id],
+          created_by_id: communityMessageFormat.user_id
         });
 
-        try {
-          // await chat.stopWatching();
-        } catch (error) {
-          console.log('::: Error on stopWatching :::', error);
+        await chat.create();
+        const channelState = await chat.watch();
+
+        if (channelState.messages.length === 0) {
+          await chat.sendMessage({
+            text: communityMessageFormat.message,
+            user_id: communityMessageFormat.user_id
+          });
+
+          try {
+            // await chat.stopWatching();
+          } catch (error) {
+            console.log('::: Error on stopWatching :::', error);
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `followTopicProcess` function, you are checking if the user is anonymous and then sending a message as an anonymous user or as a regular user. This seems fine, but again, there is no error handling for the case where any of these operations fail. It would be good to wrap this in a try-catch block and handle any potential errors that might occur during these operations.


<!-- This is an auto-generated comment by OSS CodeRabbit -->